snmp/datadog_checks/snmp/snmp.py

```diff
@@ -242,7 +284,14 @@ def check(self, instance):
         """
         # Reset errors
         self._error = self._severity = None
-        config = self._config
+        if instance.get('network_address'):
+            for discovered in list(self._config.discovered_instances.values()):
+                self._check_with_config(discovered)
+        else:
+            self._check_with_config(self._config)
```

Comment:
If we are either handling one InstanceConfig or multiples ones, should we disallow setting both `ip_address` and `network_address` at config level ?

Reply:
Yep good point.